plugin/references.py

```diff
@@ -57,37 +58,87 @@ def handle_response(self, response: 'Optional[List[Dict]]', pos) -> None:
             window.status_message("No references found")
             return
 
-        word = self.view.substr(self.view.word(pos))
+        word_region = self.view.word(pos)
+        word = self.view.substr(word_region)
 
         base_dir = get_project_path(window)
         formatted_references = self._get_formatted_references(response, base_dir)
 
-        panel = ensure_references_panel(window)
-        if not panel:
-            return
-        panel.settings().set("result_base_dir", base_dir)
-
-        panel.set_read_only(False)
-        panel.run_command("lsp_clear_panel")
-        window.run_command("show_panel", {"panel": "output.references"})
-        panel.run_command('append', {
-            'characters': "{} references for '{}'\n\n{}".format(references_count, word, formatted_references),
-            'force': True,
-            'scroll_to_end': False
-        })
-
-        # highlight all word occurrences
-        regions = panel.find_all(r"\b{}\b".format(word))
-        panel.add_regions('ReferenceHighlight', regions, 'comment', flags=sublime.DRAW_OUTLINED)
-        panel.set_read_only(True)
+        if settings.show_references_in_quick_panel:
+            window.show_quick_panel(
+                self.reflist,
+                lambda index: self.on_ref_choice(base_dir, index),
+                sublime.MONOSPACE_FONT | sublime.KEEP_OPEN_ON_FOCUS_LOST,
+                self.get_current_ref(base_dir, word_region.begin()),
```

Comment:
I was unable to figure out what `get_current_ref` does - I commented it out and passed -1 instead. Does it select a meaningful default? Otherwise I'd love to exclude that logic.

Reply:
`get_current_ref` is supposed to find the command palette entry that corresponds to the cursor position from which you sent the `find_reference` command.

So for example if you were in `plugin/core/test_rpc.py:15:20` (row 15 column 20) and sent a `find_reference` command, then `get_current_ref` should return `4` and highlight that specific entry in the command palette (`4` is the zero-based index from the list of references)

Can you try and see if that works for you? If it doesn't then every time, `find_reference` will highlight the first entry and open that specific file which is not really good if you were on another file.

Reply:
Thanks for explaining, that makes sense! 

I didn't think language servers guaranteed that a "self reference" would be included in the results. Some servers do (pyls included `types.py:11:7`, which is the declaration I invoked `find_reference` from).

In any case, you set up the opened views to be transient - so even if the self reference is missing, the user can press `ESC` and they will be where they left off.